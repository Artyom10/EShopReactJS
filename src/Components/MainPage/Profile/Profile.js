import React from 'react';

function Profile(props) {
    const {client} = props;
    return (
        <div className="container">
        <div className="row justify-content-center">
            <div className="col-md-6 text-center">
              <figure className="figure">
                  <img src={client.urlPhoto} className="figure-img img-fluid rounded profile-img" alt={client.username} />
                </figure>
              <small>  <form>
                  <div className="form-group">
                    <input type="file" className="form-control-file" id="changePhoto" />
                  </div>
                </form></small>
            </div>
        </div>
        <div className="row justify-content-center">
            <div className="col-md-6">
              <form>
                  <div className="form-group">
                    <label for="firstName">First Name</label>
                    <input type="text" className="form-control" id="firstName" value={client.firstName} />
                    <label for="secondName">Second Name</label>
                    <input type="text" className="form-control" id="secondName" value={client.secondName} />
                    <label for="secondName">Username</label>
                    <input className="form-control" type="text" value={client.username} readonly />
                    <label for="email">Email</label>
                    <input type="email" className="form-control" id="email" value={client.mail} />
                    <div className="row">
                        <div className="col-md">
                          <label for="oldPassword">Old password</label>
                          <input type="password" className="form-control" id="oldPassword" placeholder="" />
                        </div>
                        <div className="col-md">
                          <label for="newPassword">New password</label>
                          <input type="password" className="form-control" id="newPassword" placeholder="" />
                        </div>
                    </div>
                  </div>
              </form>
            </div>
        </div>
        <div className="row justify-content-center">
            <div className="col-md-6">
              <button className="btn update-button btn-block">Update</button>
              <button className="btn  general-button delete-button btn-block" data-toggle="modal" data-target="#notice">Delete request</button>
            </div>
        </div>
      </div>
    );
}

export default Profile;