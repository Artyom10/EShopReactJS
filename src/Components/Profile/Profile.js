import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import {removeRequestActionCreator} from '../../redux/reducers/clientsPageReducer';
import stylesFor from './Profile.module.css';
import { Redirect } from 'react-router-dom';

function Profile(props) {
  const {auth} = props;
  if(!auth.uid) return <Redirect to='/log_in' />
    return (
        <div className="container">
        <div className="row justify-content-center">
            <div className="col-md-6 text-center">
              <figure className="figure">
                  <img src={props.client.urlPhoto} className={`figure-img img-fluid rounded ${stylesFor.profileImg}`} alt={props.client.username} />
                </figure>
              <small>  <form>
                  <div className="form-group">
                    <input type="file" className="form-control-file" id="changePhoto" />
                  </div>
                </form></small>
            </div>
        </div>
        <div className="row justify-content-center">
            <div className="col-md-6">
              <form>
                  <div className="form-group">
                    <label for="firstName">First Name</label>
                    <input type="text" className="form-control" id="firstName" value={props.client.firstName} />
                    <label for="secondName">Second Name</label>
                    <input type="text" className="form-control" id="secondName" value={props.client.secondName} />
                    <label for="secondName">Username</label>
                    <input className="form-control" type="text" value={props.client.username} readonly />
                    <label for="email">Email</label>
                    <input type="email" className="form-control" id="email" value={props.client.mail} />
                    <div className="row">
                        <div className="col-md">
                          <label for="oldPassword">Old password</label>
                          <input type="password" className="form-control" id="oldPassword" placeholder="" />
                        </div>
                        <div className="col-md">
                          <label for="newPassword">New password</label>
                          <input type="password" className="form-control" id="newPassword" placeholder="" />
                        </div>
                    </div>
                  </div>
              </form>
            </div>
        </div>
        <div className="row justify-content-center">
            <div className="col-md-6">
              <button className={`tn ${stylesFor.updateButton} btn-block`}>Update</button>
              {props.client.request === 'False' ?
              <button className={`btn ${stylesFor.generalButton} ${stylesFor.deleteButton} btn-block`} onClick={() =>{props.removeRequest(props.client.id)}}>Delete request</button>
              : <button className={`btn ${stylesFor.generalButton} ${stylesFor.deleteButtonBack} btn-block`}>Request has been sent</button> }
            </div>
        </div>
      </div>
        
    );

}

/*
Profile.propTypes = {
  client: PropTypes.object.isRequired,
  id: PropTypes.number,
  firstName: PropTypes.string,
  secondName: PropTypes.string,
  username: PropTypes.string,
  mail: PropTypes.string,
  urlPhoto: PropTypes.string,
  password: PropTypes.string
}*/

export default Profile;