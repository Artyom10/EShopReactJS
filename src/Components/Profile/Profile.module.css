
  
  .generalButton {
    color: #fff !important;
    text-decoration: none;
    padding: 5px;
    border-radius: 5px;
    display: inline-block;
    border: none;
    transition: all 0.4s ease 0s;
    margin-bottom: 15px;
  }

  
  .generalButton:hover {
    letter-spacing: 1px;
    -webkit-box-shadow: 0px 5px 40px -10px rgba(0, 0, 0, 0.57);
    -moz-box-shadow: 0px 5px 40px -10px rgba(0, 0, 0, 0.57);
    box-shadow: 5px 40px -10px rgba(0, 0, 0, 0.57);
    transition: all 0.4s ease 0s;
  }


  .updateButton {
    
    background-color: #2796ff;
    border: none;
    box-shadow: 0 2px 8px 0 rgba(39, 150, 255, 0.6);
    letter-spacing: 0.2px;
    color: #fff;
  }


  .deleteButton {
    background: #ed3330;
  }
  
  .deleteButton-back {
    background: gold;
  }
  
  .deleteButton:hover {
    background: #434343;
  }
  
  .cancelButton {
    background: rgba(39, 150, 255, 0.6);
  }
  
  .cancelButton:hover {
    background: #434343;
  }
  
  .profileImg {
    max-width: 350px;
    max-height: 400px;
  }