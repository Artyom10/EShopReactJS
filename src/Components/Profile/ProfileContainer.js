import React from 'react';
import PropTypes from 'prop-types';
import ModalDeleteProfile from './ModalDeleteProfile/ModalDeleteProfile';
import {removeRequestActionCreator} from '../../redux/clientsPage-reducer';
import Profile from './Profile';
import { connect } from 'react-redux';


const mapStateToProp = (state) => {
    return {
     client: state.clientsPage.clients[2],
    }
 }
 
 const mapDispatchToProp = (dispatch) => {
   return {
    removeRequest: (data) => {
        let action = removeRequestActionCreator(data);
       dispatch(action);
    } 
   }
 }
 
 const ProfileContainer = connect(mapStateToProp,mapDispatchToProp)(Profile);

export default ProfileContainer;