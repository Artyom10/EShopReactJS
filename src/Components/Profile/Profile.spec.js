import { shallow } from 'enzyme';
import React from 'react';
import Profile from './Profile';


const props = {
    profile: {
        firstName: 'Name',
        secondName: 'SecondName',
        request: false,
    },
    updateProfile(){},
    removeRequest(){},
    cancelRequest(){},
    auth: {
        uid: 111,
    }
}

const setUp = (props) => shallow(<Profile {...props} />);

describe('Profile component', () => {
    it('should render Profile component', () => {
       const component = setUp(props);
        expect(component.debug()).toMatchSnapshot();
    });
});