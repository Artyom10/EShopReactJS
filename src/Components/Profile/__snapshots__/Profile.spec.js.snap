// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component should render Profile component 1`] = `
"<div className=\\"container\\">
  <div className=\\"row justify-content-center\\">
    <div className=\\"col-md-6 text-center\\">
      <figure className=\\"figure\\">
        <img src={[undefined]} className=\\"figure-img img-fluid rounded profileImg\\" />
      </figure>
    </div>
  </div>
  <div className=\\"row justify-content-center\\">
    <div className=\\"col-md-6\\">
      <form>
        <div className=\\"form-group\\">
          <label htmlFor=\\"firstName\\">
            First Name
          </label>
          <input type=\\"text\\" className=\\"form-control\\" id=\\"firstName\\" value=\\"Name\\" onChange={[Function: handleChange]} />
          <label htmlFor=\\"secondName\\">
            Second Name
          </label>
          <input type=\\"text\\" className=\\"form-control\\" id=\\"secondName\\" value=\\"SecondName\\" onChange={[Function: handleChange]} />
        </div>
      </form>
    </div>
  </div>
  <div className=\\"row justify-content-center\\">
    <div className=\\"col-md-6\\">
      <button className=\\"tn updateButton btn-block\\" onClick={[Function: handleSubmit]}>
        Update
      </button>
      <button className=\\"btn generalButton deleteButton btn-block\\" onClick={[Function: onClick]}>
        Delete request
      </button>
    </div>
  </div>
</div>"
`;
